Sources/Moya/MoyaError.swift

```diff
@@ -39,13 +39,28 @@ public extension MoyaError {
         case .jsonMapping(let response): return response
         case .stringMapping(let response): return response
         case .objectMapping(_, let response): return response
+        case .encodableMapping: return nil
         case .statusCode(let response): return response
         case .underlying(_, let response): return response
-        case .encodableMapping: return nil
         case .requestMapping: return nil
         case .parameterEncoding: return nil
         }
     }
+
+    /// Depending on error type, returns an underlying `Error`.
+    var underlyingError: Swift.Error? {
```

Comment:
I think that we should just leave this property internal. More than half of the errors do not have the underlying error inside of it so it might confuse more people than it could help.

Also, we already have a case named `underlying` in `Moya.error` and so it might be confusing as well: is the `underlyingError` an error of the underlying case or just an error associated with the case? Do you have an idea of a different name for this purpose?

Reply:
I can totally get behind leaving it `internal`. Its purpose is a lot less about providing convenient error "unwrapping" and a lot more about being a right(-ish?) way to express the mapping to `Error?` for each enum case.

Hypothetical (agreeing with the point): if one or a number of Moya's consumers needed to unwrap many different underlying `Error`s in many different ways, and were seeking out or looking to build conveniences for doing so...

1. They'd need conveniences that operate on a lot more than just `MoyaError`
2. `moyaError.underlyingError` would be a moot point compared to the far more generic and not-that-much-longer `(error as NSError).userInfo[NSUnderlyingErrorKey] as? Error`